src/content/testing/common-errors.md

```diff
@@ -475,43 +496,49 @@ framework for every frame, for example, during an animation.
 
 **How to fix it?**
 
-One way to avoid this error is to use the `Navigator` API
-to trigger the dialog as a route. In the following example,
-there are two pages. The second page has a
-dialog to be displayed upon entry.
-When the user requests the second page by
-clicking a button on the first page,
-the `Navigator` pushes two routes–one
-for the second page and another for the dialog.
+One way to avoid this error is instruct flutter to finish rendering the page
+before implementing the showDialog. This can be done by using
+addPostFrameCallback() method. The following code illustrates this on the broken example just given:
 
 <?code-excerpt "lib/set_state_build.dart (solution)"?>
 ```dart
-class FirstScreen extends StatelessWidget {
-  const FirstScreen({super.key});
+import 'package:flutter/material.dart';
+
+void main() => runApp(const ShowDialogExampleApp());
+
+class ShowDialogExampleApp extends StatelessWidget {
+  const ShowDialogExampleApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      theme: ThemeData(
+          colorSchemeSeed: const Color(0xff6750a4), useMaterial3: true),
+      home: const DialogExample(),
+    );
+  }
+}
+
+class DialogExample extends StatelessWidget {
+  const DialogExample({super.key});
 
   @override
   Widget build(BuildContext context) {
+    //You can do this:
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      showDialog<void>(
+        context: context,
+        builder: (BuildContext context) {
+          return const AlertDialog(
+            title: Text('Alert Dialog'),
+          );
+        },
+      );
+    });
     return Scaffold(
-      appBar: AppBar(
-        title: const Text('First Screen'),
-      ),
-      body: Center(
-        child: ElevatedButton(
-          child: const Text('Launch screen'),
-          onPressed: () {
-            // Navigate to the second screen using a named route.
-            Navigator.pushNamed(context, '/second');
-            // Immediately show a dialog upon loading the second screen.
-            Navigator.push(
-              context,
-              PageRouteBuilder(
-                barrierDismissible: true,
-                opaque: false,
-                pageBuilder: (_, anim1, anim2) => const MyDialog(),
-              ),
-            );
-          },
```

Comment:
Here's a possible solution that doesn't involve stateful widgets (using the Navigator's `context` for the dialog, to ensure it isn't disposed of by the end of the frame):

```dart
onPressed: () {
  final NavigatorState navigator = Navigator.of(context);
  navigator.pushNamed('/second');

  WidgetsBinding.instance.addPostFrameCallback((_) {
    showDialog(
      context: navigator.context,
      builder: (context) => const MyDialog(),
    );
  });
},
```

Reply:
The difficulty I'm having with implementing this is that my example code doesn't have an onPressed callback anywhere in it, and if I create a button in the Scaffold and put the showDialog in its onPressed:,, the "setState error during build" error disappears without the need of a addPostFrameCallback.

I'm afraid that the broken code I give as an example is crazy - nobody in their right mind would write it. The actual code that caused me the problem was with pluto_grid, and I don't know how to create a sensible example of the error without it - and I can see that pluto_grid should not be included in a page like this. I guess the solution snippet in the current version is a good solution to a problem which was also too specific - which is why I didn't find it helpful.

I suppose I've got into trouble because I'm trying to post a solution to a specific problem in a "common errors" page whose purpose has to be much more general. 

If I can't find a sensible example of broken code - which uses a callback like onPressed or onChanged and causes this particular error without using pluto_grid, I don't think I can make a sensible contribution. What do you think?

Reply:
Thanks for sharing your thoughts.

I think you made a good point about the post-frame callback not being necessary here 🙂
I edited the comment above accordingly, thanks!